egs/sad_rats/s5/run.sh

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env bash
+# Copyright   2021   ARL (Author: John Morgan)
+# Apache 2.0.
+
+# This recipe builds a Speech Activity Detection system on the rats_sad corpus.
+# The LDC identifyer for the rats_sad corpus is LDC2015S02.
+. ./cmd.sh
+. ./path.sh
+set -euo pipefail
+stage=0
+
+# Path where RATS_SAD gets downloaded (or where locally available):
+rats_sad_data_dir=/export/corpora5/LDC/LDC2015S02/data
+nj=50
+test_sets="dev-1 dev-2 "
+affix=1a
+dir=exp/sad_${affix}
+
+. utils/parse_options.sh
+
+if [ $stage -le 0 ]; then
+  # Consolidate the data from the corpus annotation files
+  echo "$0: Get  all info from annotation files."
+  local/rats_sad_tab_prep.sh $rats_sad_data_dir
+fi
+
+if [ $stage -le 1 ]; then
+  # Prepare data directories
+  for fld in train $test_sets ; do
+    echo "$0: preparing $fld data set."
+    mkdir -p data/$fld
+    local/prepare_data.py data/local/annotations/$fld.txt \
+      $rats_sad_data_dir/$fld/audio/ data/$fld
+  done
+fi
+
+if [ $stage -le 2 ]; then
+  # Write utt2spk and segments files
+  for fld in train $test_sets ; do
+    echo "$0: Convert $fld rttm files to utt2spk and segments for ${fld}."
+    local/convert_rttm_to_utt2spk_and_segments.py --use-reco-id-as-spkr=true \
+      data/$fld/rttm.annotation \
+      <(awk '{print $2" "$2" "$3}' data/$fld/rttm.annotation |sort -u) \
+      data/$fld/utt2spk data/$fld/segments
+
+    utils/utt2spk_to_spk2utt.pl data/$fld/utt2spk > data/$fld/spk2utt
+    utils/fix_data_dir.sh data/$fld
+  done
+fi
+
+if [ $stage -le 3 ]; then
+  # Extract MFCCs for training data set
+  echo "$0: Extract features for train data directory."
+  local/make_mfcc.sh --nj $nj --cmd "$train_cmd"  --write-utt2num-frames true \
```

Comment:
If you are training the SAD on the `whole` data directory (for which you extract MFCCs in stage 5), do you still need to exract the MFCCs here?

Reply:
The segment-level feature extraction step was unnecessary. I removed that stage.